irohad/ordering/on_demand_os_transport.hpp

```diff
@@ -23,32 +26,65 @@ namespace iroha {
     namespace transport {
 
       /**
-       * Type of round indexing
+       * Type of round indexing by blocks
+       */
+      using BlockRoundType = uint64_t;
+
+      /**
+       * Type of round indexing by reject before new block commit
        */
-      using RoundType = uint64_t;
+      using RejectRoundType = uint32_t;
+
+      /**
+       * Type of proposal round
+       */
+      using RoundType = std::pair<BlockRoundType, RejectRoundType>;
+
+      /**
+       * Class provides hash function for RoundType
+       */
+      class RoundTypeHasher {
+       public:
+        std::size_t operator()(const RoundType &val) const {
+          return boost::hash_value(val);
+        }
+      };
 
       /**
        * Notification interface of on demand ordering service.
        */
       class OdOsNotification {
        public:
+        /**
+         * Type of stored proposals
+         */
+        using ProposalType = std::shared_ptr<shared_model::interface::Proposal>;
```

Comment:
Please state why shared is required here in the doc. It suggests that implementation will give a copy of the pointer, which then in turn should be copied into transport object. Maybe create a copy inside ordering service and give a `unique_ptr` instead?